### PR TITLE
Fix a few more lint warnings.

### PR DESCRIPTION
**Pull request by Dominator008**
Fix goog.provide()'s that were not sorted alphabetically in closure/goog/proto2/test.pb.js.

Add missing JSDoc to functions in closure/goog/soy/soy_testhelper.js.

### CODE REVIEW

**Comment by Dominator008** [comment]
This might require an internal cleanup if some teams are still on gjslint.

**Comment by concavelenz** [comment]
I think it is worth having a conversation about whether this is something the compile based linter is going to add first.

**Comment by Dominator008** [comment]
@MatrixFrog

**Comment by MatrixFrog** [comment]
We're already having that conversation in https://github.com/google/closure-compiler/issues/1615

**Comment by Dominator008** [comment]
@MatrixFrog I think what @concavelenz meant is that whether the Compiler-based linter should actually check for "extraProvide", not just whether it should be allowed (and essentially ignored).

**Comment by Dominator008** [comment]
@concavelenz I'm adding `extraProvide` to allowed suppression types in the Compiler-based linter in https://github.com/google/closure-compiler/pull/1624. I've reduced this PR to just fixing some lint warnings.

File: closure/goog/soy/soy_testhelper.js [hunk 1]
```diff
@@ -64,44 +72,79 @@ function makeSanitizedContent(content, kind) {
 var example = {};
 
 
-example.textNodeTemplate = function(opt_data, opt_sb, opt_injectedData) {
-  assertNotNull(opt_data);
-  assertNotUndefined(opt_data);
-  return goog.string.htmlEscape(opt_data.name);
+/**
+ * @param {{name: string}} data
+ * @param {null=} opt_sb
```
Comment (joeltine) [inline closure/goog/soy/soy_testhelper.js]: Why `{null=}` here? This type wizardy scares and confuses me.

Comment (Dominator008) [inline closure/goog/soy/soy_testhelper.js]: I think it is a legacy [parameter or something.. I can't figure out its full name.

Comment (joeltine) [inline closure/goog/soy/soy_testhelper.js]: How did you determine these `@param` types?

Comment (Dominator008) [inline closure/goog/soy/soy_testhelper.js]: These are Soy templates.
`soy_test.js` uses:
`goog.soy.renderElement`: https://github.com/google/closure-library/blob/master/closure/goog/soy/soy.js#L67-L68
`goog.soy.renderAsFragment`: https://github.com/google/closure-library/blob/master/closure/goog/soy/soy.js#L91-L92
I determined the types from the signatures there.

**Pull request merged**